### DISCUSSION TITLE
How to distribute a dynamic number of items evenly in the parent control?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#14387] [Q&A] [answered]
I have a number of items that is changing over time. I want to display them evenly spaced on screen so that every item takes 1/n of the availble height. On top of that I want to limit the minimum height of an item so that the items will be stacked on top of each other with their minimum size if thei would become to small.

How can I do that?

**Comment by workgroupengineering** [comment]
Creation of your custom panel. You can start from the [UniformGrid](https://github.com/AvaloniaUI/Avalonia/blob/0bde86b4577fbf2580dbffc3892de2e06e74b95d/src/Avalonia.Base/Platform/Internal/AssemblyDescriptorResolver.cs#L41)

**Comment by timunie** [comment]
ItemsRepeater sounds like a valid option here

**Comment by DevAM-Tools** [comment, marked as answer]
This is doing the job for me:

``` Xaml
<Grid
	Name="Container"
	Bounds="{Binding ContainerBounds, Mode=OneWayToSource}">
	<ScrollViewer>
		<ItemsRepeater
			ItemsSource="{Binding Items}">
			<ItemsRepeater.ItemTemplate>
				<DataTemplate>
					
					<Border
						Classes="Border"
						Background="DarkRed"
						VerticalAlignment="Top"
						Height="{Binding Height}"
						MinHeight="{Binding MinHeight}">

						<TextBlock Text="{Binding Text}"/>
					</Border>
					
				</DataTemplate>
			</ItemsRepeater.ItemTemplate>
		</ItemsRepeater>
	</ScrollViewer>
</Grid>
```

Necessary properties of the item:

``` Csharp
public static double MinHeight => 100.0;

private double _Height;

internal double Height
{
    get => _Height;
    set
    {
        this.RaiseAndSetIfChanged(ref _Height, value);
    }
}
```

And the parent of the items:

``` Csharp

internal ObservableCollectionExtended<Item> Items { get; init; } = [];

private Rect _ContainerBounds;

internal Rect ContainerBounds
{
    get => _ContainerBounds;
    set
    {
        this.RaiseAndSetIfChanged(ref _ContainerBounds, value);

        double height = Items.Count > 0 && double.IsRealNumber(value.Height) ? value.Height / Items.Count : 0.0;
        for (int i = 0; i < Items.Count; i++)
        {
            Items[i].Height = height;
        }
    }
}
```

On top of that you need to handle the changes of the collection.

That's what it looks like: 
![grafik](https://github.com/AvaloniaUI/Avalonia/assets/125025268/f7cc258d-c0fc-4ee8-b78d-4fde34381e3c)

And that's what it looks like when the parent control does not provide enough space:
![grafik](https://github.com/AvaloniaUI/Avalonia/assets/125025268/c3759c54-c3cc-4158-8483-6ef0cfc71b62)
